crates/block-producer/src/mempool/mod.rs

```diff
@@ -231,6 +237,11 @@ impl Mempool {
             .expect("Transaction graph is malformed");
 
         // Rollback state.
+        let transactions = transactions
+            .into_iter()
+            // FIXME
+            .map(|tx_id| todo!("Inflight state should remember diffs"))
+            .collect::<Vec<_>>();
```

Comment:
I'm intending to change `InflightState` to track transaction ID -> state delta internally in a follow-up PR.